### PR TITLE
Feature/rework create office endpoint

### PR DESCRIPTION
**Pull request by SolovyovDmitry**
https://trello.com/c/rzr7sOEN

### CODE REVIEW

File: src/OfficeService/Controllers/OfficeController.cs [hunk 1]
```diff
@@ -20,21 +23,30 @@ public class OfficeController : ControllerBase
...
-      [FromServices] ICreateOfficeCommand command,
-      [FromBody] CreateOfficeRequest request)
+    public async Task<IActionResult> CreateAsync(
+      [FromBody] CreateOfficeRequest request,
+      CancellationToken ct)
     {
-      return await command.ExecuteAsync(request);
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanies))
+      {
+        //TODO: Forbid() cannot be use without authentication handler
```
Comment (SolovyovDmitry) [inline src/OfficeService/Controllers/OfficeController.cs]: Let's decide if it's ok

File: src/OfficeService/ValidationPipelineBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+using FluentValidation;
+using FluentValidation.Results;
+using LT.DigitalOffice.Kernel.Exceptions.Models;
+using MediatR;
+
+namespace LT.DigitalOffice.OfficeService
+{
+  public class ValidationPipelineBehaviour<TRequest, TResponse> : IPipelineBehavior<TRequest, TResponse> where TRequest : IRequest<TResponse>
+  {
+    //TODO: move into kernel
```
Comment (SolovyovDmitry) [inline src/OfficeService/ValidationPipelineBehaviour.cs]: I suggest this

Comment (vorobevamv) [reply src/OfficeService/ValidationPipelineBehaviour.cs]: don't mind

File: src/OfficeService/ValidationPipelineBehaviour.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    {
+      if (!_validators.Any())
+      {
+        return await next();
+      }
+
+      ValidationContext<TRequest> context = new(request);
+      ValidationResult[] validationResults = await Task.WhenAll(_validators.Select(v => v.ValidateAsync(context, cancellationToken)));
+      List<ValidationFailure> failures = validationResults.SelectMany(r => r.Errors).Where(f => f != null).ToList();
+      if (failures.Count != 0)
```
Comment (VLiamin) [inline src/OfficeService/ValidationPipelineBehaviour.cs]: May be add emty line?

File: src/OfficeService/ValidationPipelineBehaviour.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+namespace LT.DigitalOffice.OfficeService
+{
+  public class ValidationPipelineBehaviour<TRequest, TResponse> : IPipelineBehavior<TRequest, TResponse> where TRequest : IRequest<TResponse>
+  {
+    //TODO: move into kernel
+    private readonly IEnumerable<IValidator<TRequest>> _validators;
+    public ValidationPipelineBehaviour(IEnumerable<IValidator<TRequest>> validators)
+    {
+      _validators = validators;
+    }
```
Comment (VLiamin) [inline src/OfficeService/ValidationPipelineBehaviour.cs]: May be add emty line?

**Review by ndoriomedov — APPROVED** [review]

**Review by vorobevamv — APPROVED** [review]

**Pull request merged**